src/library/scala/collection/Map.scala

```diff
@@ -47,6 +47,11 @@ trait Map[K, +V]
 
   override def hashCode(): Int = MurmurHash3.mapHash(toIterable)
 
+  // These two methods are not in MapOps so that MapView is not forced to implement them
+  @deprecated("Use - or remove on an immutable Map", "2.13.0")
+  def - (key: K): Map[K, V]
+  @deprecated("Use -- or removeAll on an immutable Map", "2.13.0")
+  def - (key1: K, key2: K, keys: K*): Map[K, V]
```

Comment:
Actually, this is going to be painful for users inheriting `Map`. We should not require them to implement a deprecated operation. What do you think of implementing it with `filter`, instead?

Reply:
Nobody should extend `collection.Map`. You either extend `collection.mutable.Map` or `collection.immutable.Map`, both of which implement the methods.

Reply:
This is precisely what happened in `pos/t247.scala` but that's a test case, not real-world code. IMHO the performance characteristics of a `filter`-based default implementation are so surprising that I'd rather make it an error if you don't implement the methods.